python/rapidsmpf/rapidsmpf/buffer/resource.pyx

```diff
@@ -255,6 +267,7 @@ cdef class BufferResource:
                 move(_mem_available),
                 period,
                 cpp_stream_pool,
+                (<Statistics>statistics)._handle,
```

Comment:
```suggestion
                (<Statistics?>statistics)._handle,
```

Reply:
This doesn't compile, I guess because we're inside a `with nogil` block. I assigned the casted handle to a local earlier, when we do have the GIL.